clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -6664,6 +6664,13 @@ def err_builtin_counted_by_ref_invalid_lhs_use : Error<
 def err_builtin_counted_by_ref_has_side_effects : Error<
   "'__builtin_counted_by_ref' argument cannot have side-effects">;
 
+def warn_wraps_attr_var_decl_type_not_integer : Warning<
+  "using attribute '%select{wraps|no_wraps}0' with non-integer type '%1' has no function and is potentially misleading">,
+  InGroup<UselessWrapsAttr>;
```

Comment:
how about dropping "has no function and" so the final sentence becomes something like:

`"using attribute 'wraps' with non-integer type '%1' is potentially misleading"`

Reply:
something more like:
`attribute 'wraps' has no effect with non-integer type 'float'; attribute ignored` ?

Reply:
since this is now an error ([3de3561](https://github.com/llvm/llvm-project/pull/115094/commits/3de3561f6c6e4e73ae1f5cda3a27a6aff5bf9904)) we can be a bit more assertive with the language. I went with:

`cannot use attribute 'wraps' with non-integer type 'float'` which is an error when parsing.